ballerina-tests/tests/43_id_input_type_validation.bal

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config {
+    groups: ["id_validation"],
+    dataProvider: dataProviderIdAnnotation
+}
+isolated function testIdTypeAnnotation(string url, string resourceFileName) returns error? {
+    string document = check getGraphqlDocumentFromFile(resourceFileName);
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile(resourceFileName);
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+function dataProviderIdAnnotation() returns map<[string, string]> {
+    string url = "http://localhost:9091/id_annotation_2";
```

Comment:
seems url doesn't change here, we could move the url to the testFunction and remove the url param.